### ISSUE TITLE
ie11 support

### ISSUE THREAD

**Issue by brandynbennett** [microstates/ember#115]
When trying to use the addon in ie11 our app doesn't load. It looks like `index.js` is importing `microstates/dist/microstates.cjs` which isn't transpiled.

**Comment by knownasilya** [comment]
I think everything should go through babel, so if your `config/targets.js` includes `ie11` then this should be transpiled too. I think..

**Comment by brandynbennett** [comment]
We have ie11 in our targets, but es6+ syntax is showing up in our vendor.js from microstates.

**Comment by brandynbennett** [comment]
May be along a similar vain as https://github.com/ef4/ember-auto-import/issues/41

**Comment by brandynbennett** [comment]
I tried adding an ember-auto-import config to point `microstates` to the `microstates.es.js` file instead. This fixes our app, but then one of the addon tests breaks:

`allows to create a Microstate<Person> using (type "person")`
throws an error: `TypeError: Class constructor Person cannot be invoked without 'new'`

**Comment by brandynbennett** [comment]
Ok seems like this is because the test/dummy app are using es6+ code and the microstates.js library is now using es5 code so microstates is trying to instantiate a class without `new` because that's how you'd do it in es5 land.

**Comment by brandynbennett** [comment]
Seems related to https://github.com/microstates/microstates.js/issues/136 as well

**Comment by brandynbennett** [comment]
Can't use the UMD version because it only transpiles to Chrome 58 https://github.com/microstates/microstates.js/blob/master/rollup.config.js#L54

**Comment by knownasilya** [comment]
@brandynbennett can you comment on microstates/microstates.js#136 to move it back to the top.

**Comment by taras** [comment]
Yeah, this an irritating part the current situation with transpiration and libraries. Trying to figure this out has been like juggling compromises. I'll comment in 
https://github.com/microstates/microstates.js/issues/136 to keep this conversation going.

**Comment by taras** [comment]
@brandynbennett it looks like there might be a solution in the works for this in ember-auto-import. Once that work is complete we'd need to change this library to use the new API that `ember-auto-import` would provide. 

In the mean time, if this problem is blocking you from using microstates for a production app, then we might need to find a temporary work around. The work around might be to detect the babel target and automatically include a different build. 

Let me know if this is blocking you. If it is, I'll prioritize it to get it fixed ASAP.

**Comment by brandynbennett** [comment]
Thanks @taras. We found a hacky workaround which involved mutating the imported file source in `addon/index.js` during the npm `postinstall` hook that got us unblocked. Here it is if it helps others while we wait for `ember-auto-import` solution
```
// package.json
scripts: {
  "postinstall": "sed -i'' -e 's|dist/microstates.cjs|dist/microstates.es|g' node_modules/@microstates/ember/addon/index.js"
}
```

**Comment by brandynbennett** [comment]
@taras I went ahead and made a PR https://github.com/microstates/ember/pull/117 that dynamically pick which build to use based on targets.

**Comment by taras** [comment]
Released in 0.12.3

Thank you!

**Comment by brandynbennett** [comment]
Thanks so much for the quick release!
